lib/api/expect/assertions/_baseAssertion.js

```diff
@@ -347,7 +348,7 @@ class BaseAssertion {
   }
 
   shouldRetry() {
-    if (!this.waitForMs || this.emitter.retryUnavailable) {
+    if (!this.waitForMs || this.emitter.retryUnavailable || this.retries > this.maximumRetries) {
```

Comment:
@beatfactor there was no way to deterministically limit the number of retries.. so I added this flag. Thoughts?

Reply:
I don't think we should change anything in the main codebase in this PR. This is purely fixing the exit code issue and if it turns out that some tests are unstable, we need to rewrite the tests.

Reply:
okay, reverted.